pkg/apis/monitoring/v1/types.go

```diff
@@ -705,6 +705,19 @@ type AttachMetadata struct {
 	//
 	// +optional
 	Node *bool `json:"node,omitempty"`
+
+	// namespace when set to true, Prometheus attaches namespace metadata to the discovered
+	// targets.
+	//
+	// The Prometheus service account must have the `list` and `watch`
+	// permissions on the `Namespaces` objects.
+	//
+	// Only valid for pod, endpoints, endpointslice, service, ingress.
```

Comment:
we don't need the comment since it's "only" supported for ServiceMonitor/PodMonitor/Probe?